Formula/snappy.rb

```diff
@@ -17,11 +17,35 @@ class Snappy < Formula
   depends_on "cmake" => :build
   depends_on "pkg-config" => :build
 
+  # Fix issue where Mojave clang fails to build due to entering a __GNUC__ block
+  on_macos do
+    depends_on "llvm" if DevelopmentTools.clang_build_version <= 1100
+  end
+
+  fails_with :clang do
+    build 1100
+    cause "error: invalid output constraint '=@ccz' in asm"
+  end
+
+  # Fix issue where `snappy` setting -fno-rtti causes build issues on `folly`
+  # `folly` issue ref: https://github.com/facebook/folly/issues/1583
+  patch :DATA
+
   def install
-    system "cmake", ".", *std_cmake_args
+    on_macos do
```

Comment:
Actually, we should consider adding an
```
ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_bin
```
here so that `llvm` can be a build-dependency instead of a runtime one. (See, e.g, `grpc`, `deno`, `v8`)